pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -231,17 +235,28 @@ func newOpenStack(cfg Config) (*OpenStack, error) {
 		provider.HTTPClient.Transport = netutil.SetOldTransportDefaults(&http.Transport{TLSClientConfig: config})
 
 	}
-	if cfg.Global.TrustId != "" {
+
+	authUrl := cfg.Global.AuthUrl
+	keystone_version := "Unknow Keystone Version"
+	if strings.Contains(authUrl, Keystone_v2) {
+		keystone_version = Keystone_v2
+	} else if strings.Contains(authUrl, Keystone_v3) {
+		keystone_version = Keystone_v3
+	}
+
+	switch keystone_version {
```

Comment:
I dont think this is needed. Authenticate(...) has already done the version choose work.